polar-core/src/polar.rs

```diff
@@ -36,6 +37,10 @@ impl Query {
     pub fn next_message(&self) -> Option<Message> {
         self.vm.messages.next()
     }
+
+    pub fn source(&self) -> String {
+        self.vm.term_source(&self.source)
```

Comment:
Maybe we should include the context too? From `source.source`? Will have line number for example.